src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryControl.cs

```diff
@@ -651,10 +841,12 @@ public SortRequestControl(params SortKey[] sortKeys) : base("1.2.840.113556.1.4.
                 }
             }
 
+            _keysAsnLength = 0;
             _keys = new SortKey[sortKeys.Length];
             for (int i = 0; i < sortKeys.Length; i++)
             {
                 _keys[i] = new SortKey(sortKeys[i].AttributeName, sortKeys[i].MatchingRule, sortKeys[i].ReverseOrder);
+                _keysAsnLength += 13 + (sortKeys[i].AttributeName?.Length ?? 0) + (sortKeys[i].MatchingRule?.Length ?? 0);
```

Comment:
I could ask "what's 13?", then say "consider adding a comment", but my real answer is "stop doing manual size calculations".